pkg/storage/image.go

```diff
@@ -271,11 +282,44 @@ func (svc *imageService) isSecureIndex(indexName string) bool {
 	return true
 }
 
+func isValidHostname(hostname string) bool {
+	return hostname != "" && !strings.Contains(hostname, "/") &&
+		(strings.Contains(hostname, ".") ||
+			strings.Contains(hostname, ":") || hostname == "localhost")
+}
+
+func (svc *imageService) ResolveNames(imageName string) []string {
+	domain, rest := splitDomain(imageName)
+	if len(domain) != 0 && isValidHostname(domain) {
+		// this means the image is already fully qualified
+		return []string{imageName}
+	}
+	// this means we got an image in the form of "busybox"
+	// we need to use additional registries...
+	// normalize the unqualified image to be domain/repo/image...
+	images := []string{}
+	for _, r := range svc.registries {
+		if r == dockerRegistry {
```

Comment:
I would rather remove the reference to docker.io and allow containers/image to do its thing.  You are building in assumptions about containers/image that could change in the future.  I think we should treat them all the same.